File: CSHTML5.Samples.Showcase/Samples/XAML_Features/Binding2/KilometersToMilesConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml.Data;

namespace CSHTML5.Samples.Showcase
{
    public class KilometersToMilesConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            int kilometers;
            if (value != null && int.TryParse(value.ToString(), out kilometers))
            {
                return (int)(kilometers * 0.62137119);
            }
            else
                return "";
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            int miles;
            if (value != null && int.TryParse(value.ToString(), out miles))
            {
                return (int)(miles * 1.6093440);
            }
            else
                return "";
        }
    }
}

File: src/Samples/XAML_Features/Binding2/KilometersToMilesConverter.fs
namespace OpenSilver.Samples.Showcase

open System
open System.Windows.Data
open System.Globalization

type KilometersToMilesConverter() =
    interface IValueConverter with
        member this.Convert(value, targetType, parameter, culture) =
            let mutable result: obj = ""
            match value with
            | null -> result
            | _ ->
                let mutable kilometers = 0
                if Int32.TryParse(value.ToString(), &kilometers) then
                    result <- (int)((float)kilometers * 0.62137119)
                result

        member this.ConvertBack(value, targetType, parameter, culture) =
            let mutable result: obj = ""
            match value with
            | null -> result
            | _ ->
                let mutable miles = 0
                if Int32.TryParse(value.ToString(), &miles) then
                    result <- (int)((float)miles * 1.6093440)
                result
